### PR TITLE
Added tables for contest samples

### LINKED ISSUE #2152: Add tables for contest samples

### PR DESCRIPTION
**Pull request by Cheshiriks**
The user can create a template for a specific future contest. He will need to describe the fields that are required to be filled in by the participants of this contest.

The ContestSampleField table will contain a description of each specific field for the contest.

**what's done**

- Added tables for contest sample and contest sample fields
- Added entities and dto for contest sample
- moved all dto for contest in contest folder

Closes #2152

### CODE REVIEW

**Comment by akuleshov7** [comment]
> Closes #2152 

Please add "what's done" description and why do we need it

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/ContestSampleField.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import javax.persistence.*
+
+/**
+ * @property contestSample contest sample
+ * @property name name of field
+ * @property type type of field
+ * @property userId creator contestSampleField
+ */
+@Entity
+class ContestSampleField(
```
Comment (akuleshov7) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/ContestSampleField.kt:15]: Contest sample field - is not the best name for this entity. I do not understand why you call it "поле экземпляра контеста"

**Review by akuleshov7 — APPROVED** [review]
Lgtm, but need better description. Also would be great if you will add comments in code to entities: why do we need them and how do we use them

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/ContestSampleField.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/ContestSampleField.kt:15]: The user can create a template for a specific future contest. He will need to describe the fields that are required to be filled in by the participants of this contest.

The ContestSampleField table will contain a description of each specific field for the contest.

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/contest/ContestSampleDto.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import kotlinx.serialization.Serializable
+
+/**
+ * @property name name of contest sample
+ * @property description description of contest sample
+ */
+@Serializable
+data class ContestSampleDto(
+    val name: String,
+    val description: String?,
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/contest/ContestSampleDto.kt:12]: Changed it

**Pull request merged**